rdsa_utils/cdp/io/output.py

```diff
@@ -65,36 +92,77 @@ def insert_df_to_hive_table(
     ValueError
         If the SparkDF schema does not match the Hive table schema and
         'fill_missing_cols' is set to False.
+    DataframeEmptyError
+        If input DataFrame is empty.
     Exception
         For other general exceptions when writing data to the table.
-    """
-    logger.info(f"Preparing to write data to {table_name}.")
 
-    # Validate SparkDF before writing
-    if is_df_empty(df):
-        msg = f"Cannot write an empty SparkDF to {table_name}"
-        raise DataframeEmptyError(
-            msg,
-        )
+    Examples
+    --------
+    Write a DataFrame to a Hive table without overwriting:
+    >>> insert_df_to_hive_table(
+    ...     spark=spark,
+    ...     df=df,
+    ...     table_name="my_database.my_table"
+    ... )
+
+    Overwrite an existing table with a DataFrame:
+    >>> insert_df_to_hive_table(
+    ...     spark=spark,
+    ...     df=df,
+    ...     table_name="my_database.my_table",
+    ...     overwrite=True
+    ... )
+
+    Write a DataFrame to a Hive table with missing columns filled:
+    >>> insert_df_to_hive_table(
+    ...     spark=spark,
```

Comment:
worth adding what the missing cols are filled with. I.E do you use a blank string for a string column, or a NULL etc. Are missing int columns filled with defaults of 0 or NaNs?

Reply:
Added to docstring regarding this and made the logic better for fill_missing_col, it will match df column to underlying Hive Schema